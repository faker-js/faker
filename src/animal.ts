import type { Faker } from '.';

export class Animal {
  constructor(private readonly faker: Faker) {
    // Bind `this` so namespaced is working correctly
    for (const name of Object.getOwnPropertyNames(Animal.prototype)) {
      if (name === 'constructor' || typeof this[name] !== 'function') {
        continue;
      }
      this[name] = this[name].bind(this);
    }
  }

  /**
   * Returns a random dog breed.
   *
   */
  dog(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.dog);
  }

  /**
   * Returns a random cat breed.
   *
   */
  cat(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.cat);
  }

  /**
   * Returns a random snake species.
   *
   */
  snake(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.snake);
  }

  /**
   * Returns a random bear species.
   *
   */
  bear(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.bear);
  }

  /**
   * Returns a random lion species.
   *
   */
  lion(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.lion);
  }

  /**
   * Returns a random cetacean species.
   *
   * @method faker.animal.cetacean
   */
  cetacean(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.animal.cetacean
    );
  }

  /**
   * Returns a random horse breed.
   *
   */
  horse(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.horse);
  }

  /**
   * Returns a random bird species.
   *
   */
  bird(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.bird);
  }

  /**
   * Returns a random cow species.
   *
   */
  cow(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.cow);
  }

  /**
   * Returns a random fish species.
   *
   */
  fish(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.fish);
  }

  /**
   * Returns a random crocodilian species.
   *
   */
  crocodilia(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.animal.crocodilia
    );
  }

  /**
   * Returns a random insect species.
   *
   */
  insect(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.insect);
  }

  /**
   * Returns a random rabbit species
   *
   */
  rabbit(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.rabbit);
  }

  /**
   * Returns a random animal type.
   *
   */
  type(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.type);
  }
}
