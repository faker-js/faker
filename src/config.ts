/**
 * The possible configuration options, that can be set.
 * This type exists to be extended for plugins via type augmentation.
 *
 * The `@default` tag is used to indicate the default value, that should be used if, the config is absent.
 */
export interface FakerConfig {
  /**
   * The function used to generate the `refDate` date instance, if not provided as method param.
   * The function must return a new valid `Date` instance for every call.
   *
   * @see [Reproducible Results](https://fakerjs.dev/guide/usage.html#reproducible-results)
   * @see faker.seed(): For generating reproducible values.
   *
   * @since 9.0.0
   *
   * @default () => new Date()
   */
  defaultRefDate?: () => Date;
}
