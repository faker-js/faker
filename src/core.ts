import type { FakerConfig } from './config';
import type { LocaleDefinition } from './definitions';
import type { Randomizer } from './randomizer';

/**
 * The core grants access to the locale data, the randomizer and config settings.
 */
export interface FakerCore {
  /**
   * The locale data associated with this instance.
   *
   * Always present, but it might be empty if the locale data is not available.
   */
  readonly locale: LocaleDefinition;

  /**
   * The randomizer used to generate random values.
   */
  readonly randomizer: Randomizer;

  /**
   * The configuration settings used by this instance.
   */
  readonly config: FakerConfig;
}
