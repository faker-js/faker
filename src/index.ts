import { Faker } from './faker';
import allLocales from './locales';

export type {
  AnimalDefinitions,
  ColorDefinitions,
  CommerceDefinitions,
  CommerceProductNameDefinitions,
  CompanyDefinitions,
  DatabaseDefinitions,
  DateDefinitions,
  DateEntryDefinition,
  FinanceCurrencyEntryDefinitions,
  FinanceDefinitions,
  HackerDefinitions,
  InternetDefinitions,
  LocaleDefinition,
  /** @deprecated Use LocationDefinitions instead */
  LocationDefinitions as AddressDefinitions,
  LocationDefinitions,
  LoremDefinitions,
  MusicDefinitions,
  /** @deprecated Use PersonDefinitions instead */
  PersonDefinitions as NameDefinitions,
  PersonDefinitions,
  /** @deprecated Use PersonTitleDefinitions instead */
  PersonTitleDefinitions as NameTitleDefinitions,
  PersonTitleDefinitions,
  PhoneNumberDefinitions,
  ScienceDefinitions,
  SystemDefinitions,
  SystemMimeTypeEntryDefinitions,
  VehicleDefinitions,
  WordDefinitions,
} from './definitions';
export { FakerError } from './errors/faker-error';
export type { FakerOptions, UsableLocale, UsedLocales } from './faker';
export type { AnimalModule } from './modules/animal';
export type {
  Casing,
  ColorFormat,
  ColorModule,
  CSSFunction,
  CSSSpace,
  NumberColorFormat,
  StringColorFormat,
} from './modules/color';
export type { CommerceModule } from './modules/commerce';
export type { CompanyModule } from './modules/company';
export type { DatabaseModule } from './modules/database';
export type { DatatypeModule } from './modules/datatype';
export type { DateModule } from './modules/date';
export type { FinanceModule } from './modules/finance';
export type { GitModule } from './modules/git';
export type { HackerModule } from './modules/hacker';
export type { HelpersModule } from './modules/helpers';
export type { ImageModule } from './modules/image';
export type { InternetModule } from './modules/internet';
export type {
  /** @deprecated Use LocationModule instead */
  LocationModule as AddressModule,
  LocationModule,
} from './modules/location';
export type { LoremModule } from './modules/lorem';
export type { MusicModule } from './modules/music';
export { Sex } from './modules/person';
export type {
  /** @deprecated Use PersonModule instead */
  PersonModule as NameModule,
  PersonModule,
  SexType,
} from './modules/person';
export type { PhoneModule } from './modules/phone';
export type { RandomModule } from './modules/random';
export type { ChemicalElement, ScienceModule, Unit } from './modules/science';
export type { StringModule } from './modules/string';
export type { SystemModule } from './modules/system';
export type { VehicleModule } from './modules/vehicle';
export type { WordModule } from './modules/word';
export { Faker };

// since we are requiring the top level of faker, load all locales by default
export const faker: Faker = new Faker({
  locales: allLocales,
});
