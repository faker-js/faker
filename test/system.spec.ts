import validator from 'validator';
import { afterEach, describe, expect, it } from 'vitest';
import { faker } from '../src';
import { seededRuns, seededTests } from './support/seededRuns';
import { times } from './support/times';

const NON_SEEDED_BASED_RUN = 5;

const functionNames = [
  'commonFileExt',
  'commonFileName',
  'commonFileType',
  'cron',
  'directoryPath',
  'fileExt',
  'fileName',
  'filePath',
  'fileType',
  'mimeType',
  'networkInterface',
  'semver',
];

describe('system', () => {
  afterEach(() => {
    faker.locale = 'en';
  });

  seededTests(faker, 'system', (t) => {
    t.itEach(
      'commonFileExt',
      'commonFileType',
      'directoryPath',
      'filePath',
      'fileType',
      'mimeType',
      'semver'
    );

    t.describe('fileName', (t) => {
      t.it('noArgs').it('with extensionCount', { extensionCount: 2 });
    });

    t.describe('commonFileName', (t) => {
      t.it('noArgs').it('with extension', 'ext');
    });

    t.describe('fileExt', (t) => {
      t.it('noArgs').it('with mimeType', 'application/json');
    });

    t.describe('networkInterface', (t) => {
      t.it('noArgs');
      for (const interfaceSchema of [
        undefined,
        'index',
        'slot',
        'mac',
        'pci',
      ] as const) {
        for (const interfaceType of [undefined, 'en', 'wl', 'ww'] as const) {
          t.it(`with ${JSON.stringify({ interfaceType, interfaceSchema })}`, {
            interfaceType,
            interfaceSchema,
          });
        }
      }
    });

    t.describe('cron', (t) => {
      t.it('noArgs')
        .it('with includeYear true', { includeYear: true })
        .it('with includeYear false', { includeYear: false })
        .it('with includeNonStandard true', { includeNonStandard: true })
        .it('with includeNonStandard false', { includeNonStandard: false });
    });
  });

  for (const seed of seededRuns) {
    describe(`seed: ${seed}`, () => {
      for (const functionName of functionNames) {
        it(`${functionName}()`, () => {
          faker.seed(seed);

          const actual = faker.system[functionName]();

          expect(actual).toMatchSnapshot();
        });
      }
    });
  }

  describe(`random seeded tests for seed ${faker.seed()}`, () => {
    for (let i = 1; i <= NON_SEEDED_BASED_RUN; i++) {
      describe('commonFileExt()', () => {
        it('should return common file types', () => {
          const fileExt = faker.system.commonFileExt();
          const extList = [
            'gif',
            'htm',
            'html',
            'jpe',
            'jpeg',
            'jpg',
            'm1v',
            'm2a',
            'm1v',
            'm2v',
            'm3a',
            'mp2',
            'mp2a',
            'mp3',
            'mp4',
            'mp4v',
            'mpe',
            'mpeg',
            'mpg',
            'mpg4',
            'mpga',
            'pdf',
            'png',
            'shtml',
            'wav',
          ];

          expect(
            extList,
            `generated common file ext should be one of [${extList.join(
              ', '
            )}]. Got "${fileExt}".`
          ).include(fileExt);
        });
      });

      describe('commonFileName()', () => {
        it('should return common file name without system path separators', () => {
          const commonFileName = faker.system.commonFileName();

          expect(
            commonFileName,
            'generated common file name should not have path separators'
          ).not.toContain('/');
        });

        it('should return common file name with ext on the end', () => {
          const fileName = faker.system.commonFileName();

          expect(
            fileName,
            'generated common file name should have a extension'
          ).toContain('.');
        });

        it('should return common file name with given ext', () => {
          const fileName = faker.system.commonFileName('txt');

          expect(
            fileName,
            'generated common file name should not have path separators'
          ).not.toContain('/');
          expect(
            fileName,
            'generated common file name should have given ext'
          ).toContain('txt');
        });
      });

      describe('commonFileType()', () => {
        it('should return common file types', () => {
          const fileType = faker.system.commonFileType();
          const fileTypes = ['application', 'audio', 'image', 'text', 'video'];

          expect(
            fileTypes,
            `generated common file type should contain one of [${fileTypes.join(
              ','
            )}]. Got "${fileType}".`
          ).toContain(fileType);
        });
      });

      describe('directoryPath()', () => {
        it('should return unix fs directory full path', () => {
          const directoryPath = faker.system.directoryPath();

          expect(
            directoryPath.startsWith('/'),
            'generated directoryPath should start with /'
          ).toBeTruthy();
        });
      });

      describe('fileExt()', () => {
        it('should return file ext', () => {
          const fileExt = faker.system.fileExt();

          expect(
            fileExt.length,
            'generated fileExt should start with ."'
          ).toBeGreaterThan(1);
        });

        it('should return file ext based on mimeType', () => {
          const fileExt = faker.system.fileExt('text/plain');
          const extList = [
            'txt',
            'text',
            'conf',
            'def',
            'list',
            'log',
            'in',
            'ini',
          ];

          expect(
            extList,
            `generated common file ext should be one of [${extList.join(
              ','
            )}]. Got "${fileExt}".`
          ).include(fileExt);
        });
      });

      describe('fileName()', () => {
        it('should return filenames without system path separators', () => {
          const fileName = faker.system.fileName();

          expect(
            fileName.startsWith('/'),
            'generated fileNames should not have path separators'
          ).toBeFalsy();
        });

        it('should return filenames with ext on the end', () => {
          const fileName = faker.system.fileName();

          expect(
            fileName,
            'generated fileNames should have an extension'
          ).toContain('.');
        });

        it('should return filenames with 1 ext per default', () => {
          const fileName = faker.system.fileName();
          const parts = fileName.split('.');

          expect(parts).length(2);
        });

        it('should return filenames without an extension when extensionCount is 0', () => {
          const fileName = faker.system.fileName({
            extensionCount: 0,
          });

          expect(fileName).not.toContain('.');
        });

        it('should return filenames without an extension when extensionCount is negative', () => {
          const fileName = faker.system.fileName({
            extensionCount: -1,
          });

          expect(fileName).not.toContain('.');
        });

        it.each(times(10))(
          'should return filenames with %s extensions',
          (extensionCount) => {
            const fileName = faker.system.fileName({
              extensionCount,
            });
            const parts = fileName.split('.');

            expect(parts).length(extensionCount + 1);
          }
        );
      });

      describe('filePath()', () => {
        it('should return unix fs file full path', () => {
          const filePath = faker.system.filePath();
          const parts = filePath.split('/');

          expect(
            filePath.startsWith('/'),
            'generated filePath should start with /'
          ).toBeTruthy();
          expect(
            parts[parts.length - 1],
            'generated filePath should have a file extension'
          ).toMatch(/^\w+\.\w+$/);
        });
      });

      describe('mimeType()', () => {
        it('should return mime types', () => {
          const mimeType = faker.system.mimeType();

          expect(
            mimeType,
            `generated mime types should be valid mime types.`
          ).toSatisfy(validator.isMimeType);
        });
      });

      describe('semver()', () => {
        it('should return semver', () => {
          expect(
            faker.system.semver(),
            `generated semver, first number should be between 0 and 9.`
          ).toSatisfy(validator.isSemVer);
        });
      });

      describe('networkInterface()', () => {
        it('should return network interface', () => {
          const networkInterface = faker.system.networkInterface();

          expect(
            networkInterface,
            `generated network interface should be valid network interface.`
          ).toMatch(
            /^(?:P\d)?(?:en|wl|ww)(?:o\d|s\d(?:f\d)?(?:d\d)?|x[a-f\d]{12}|p\ds\d(?:f\d)?(?:d\d)?)$/
          );
        });

        it('should return a network interface with a given type', () => {
          const networkInterface = faker.system.networkInterface({
            interfaceType: 'wl',
          });

          expect(
            networkInterface,
            `generated network interface should be valid network interface.`
          ).toMatch(
            /^(?:P\d)?wl(?:o\d|s\d(?:f\d)?(?:d\d)?|x[a-f\d]{12}|p\ds\d(?:f\d)?(?:d\d)?)$/
          );
        });

        it('should return a network interface with an index schema', () => {
          const networkInterface = faker.system.networkInterface({
            interfaceSchema: 'index',
          });

          expect(
            networkInterface,
            `generated network interface should be valid network interface.`
          ).toMatch(/^(?:en|wl|ww)o\d$/);
        });

        it('should return a network interface with a slot schema', () => {
          const networkInterface = faker.system.networkInterface({
            interfaceSchema: 'slot',
          });

          expect(
            networkInterface,
            `generated network interface should be valid network interface.`
          ).toMatch(/^(?:en|wl|ww)s\d(?:f\d)?(?:d\d)?$/);
        });

        it('should return a network interface with a mac schema', () => {
          const networkInterface = faker.system.networkInterface({
            interfaceSchema: 'mac',
          });

          expect(
            networkInterface,
            `generated network interface should be valid network interface.`
          ).toMatch(/^(?:en|wl|ww)x[a-f\d]{12}$/);
        });

        it('should return a network interface with a pci schema', () => {
          const networkInterface = faker.system.networkInterface({
            interfaceSchema: 'pci',
          });

          expect(
            networkInterface,
            `generated network interface should be valid network interface.`
          ).toMatch(/^(?:P\d)?(?:en|wl|ww)p\ds\d(?:f\d)?(?:d\d)?$/);
        });

        it('should return a network interface with a given type and schema', () => {
          const networkInterface = faker.system.networkInterface({
            interfaceType: 'en',
            interfaceSchema: 'mac',
          });

          expect(
            networkInterface,
            `generated network interface should be valid network interface.`
          ).toMatch(/^enx[a-f\d]{12}$/);
        });
      });

      describe('cron()', () => {
        const regex =
          /^([1-9]|[1-5]\d|\*) ([0-9]|1\d|2[0-3]|\*) ([1-9]|[12]\d|3[01]|\*|\?) ([1-9]|1[0-2]|\*) ([0-6]|\*|\?|[A-Z]{3}) ((19[7-9]d)|20\d{2}|\*)?/;

        const regexElements = regex.toString().replace(/\//g, '').split(' ');

        it.each([
          [{}, 5],
          [{ includeYear: false }, 5],
          [{ includeYear: true }, 6],
        ])(
          'should return cron expression with correct number of valid elements - %o, %d',
          (options, count: number) => {
            const cron = faker.system.cron(options).split(' ');
            expect(cron).toHaveLength(count);
            cron.forEach((cronElement, i) =>
              expect(
                cronElement,
                `generated cron, ${cronElement} should match regex ${regexElements[i]}`
              ).toMatch(new RegExp(regexElements[i]))
            );
          }
        );

        it('should return non-standard cron expressions', () => {
          const validResults = ['1', '2', '3', '5', '*', '@'];
          expect(
            faker.system.cron({ includeNonStandard: true })[0],
            'generated cron, string should contain non-standard cron labels'
          ).toSatisfy(
            (value) => !!validResults.find((result) => value === result)
          );
        });
      });
    }
  });

  describe('extra tests', () => {
    describe('commonFileName()', () => {
      afterEach(() => {
        faker.locale = 'en';
      });

      it('#770', () => {
        faker.seed(5423027051750305);
        faker.setLocale('sk');
        faker.system.commonFileName('xml');
        faker.system.commonFileName('xml');
      });
    });
  });
});
