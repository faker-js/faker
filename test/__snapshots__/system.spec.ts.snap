// Vitest Snapshot v1

exports[`system > 42 > commonFileExt 1`] = `"png"`;

exports[`system > 42 > commonFileName > noArgs 1`] = `"lavender_shoes.mpe"`;

exports[`system > 42 > commonFileName > with extension 1`] = `"lavender_shoes.ext"`;

exports[`system > 42 > commonFileType 1`] = `"audio"`;

exports[`system > 42 > cron > noArgs 1`] = `"* 19 * 3 5"`;

exports[`system > 42 > cron > with includeNonStandard false 1`] = `"* 19 * 3 5"`;

exports[`system > 42 > cron > with includeNonStandard true 1`] = `"* 19 * 3 5"`;

exports[`system > 42 > cron > with includeYear false 1`] = `"* 19 * 3 5"`;

exports[`system > 42 > cron > with includeYear true 1`] = `"* 19 * 3 5 2047"`;

exports[`system > 42 > directoryPath 1`] = `"/opt/bin"`;

exports[`system > 42 > fileExt > noArgs 1`] = `"lrm"`;

exports[`system > 42 > fileExt > with mimeType 1`] = `"json"`;

exports[`system > 42 > fileName > noArgs 1`] = `"lavender_shoes.jxsc"`;

exports[`system > 42 > fileName > with extensionCount 1`] = `"lavender_shoes.jxsc.jardiff"`;

exports[`system > 42 > filePath 1`] = `"/opt/bin/comeback_neptunium_hip.ptid"`;

exports[`system > 42 > fileType 1`] = `"image"`;

exports[`system > 42 > mimeType 1`] = `"application/vnd.ibm.rights-management"`;

exports[`system > 42 > networkInterface > noArgs 1`] = `"wlp1s7"`;

exports[`system > 42 > networkInterface > with {"interfaceSchema":"index"} 1`] = `"wlo7"`;

exports[`system > 42 > networkInterface > with {"interfaceSchema":"mac"} 1`] = `"wlxcf2bc9927210"`;

exports[`system > 42 > networkInterface > with {"interfaceSchema":"pci"} 1`] = `"wlp9s1"`;

exports[`system > 42 > networkInterface > with {"interfaceSchema":"slot"} 1`] = `"wls7d7"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"en","interfaceSchema":"index"} 1`] = `"eno3"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"en","interfaceSchema":"mac"} 1`] = `"enx5cf2bc992721"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"en","interfaceSchema":"pci"} 1`] = `"P7enp9s1"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"en","interfaceSchema":"slot"} 1`] = `"ens3"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"en"} 1`] = `"ens7d7"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"wl","interfaceSchema":"index"} 1`] = `"wlo3"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"wl","interfaceSchema":"mac"} 1`] = `"wlx5cf2bc992721"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"wl","interfaceSchema":"pci"} 1`] = `"P7wlp9s1"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"wl","interfaceSchema":"slot"} 1`] = `"wls3"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"wl"} 1`] = `"wls7d7"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"ww","interfaceSchema":"index"} 1`] = `"wwo3"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"ww","interfaceSchema":"mac"} 1`] = `"wwx5cf2bc992721"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"ww","interfaceSchema":"pci"} 1`] = `"P7wwp9s1"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"ww","interfaceSchema":"slot"} 1`] = `"wws3"`;

exports[`system > 42 > networkInterface > with {"interfaceType":"ww"} 1`] = `"wws7d7"`;

exports[`system > 42 > networkInterface > with {} 1`] = `"wlp1s7"`;

exports[`system > 42 > semver 1`] = `"3.7.9"`;

exports[`system > 1211 > commonFileExt 1`] = `"htm"`;

exports[`system > 1211 > commonFileName > noArgs 1`] = `"invoice_cyclocross_assault.mp4"`;

exports[`system > 1211 > commonFileName > with extension 1`] = `"invoice_cyclocross_assault.ext"`;

exports[`system > 1211 > commonFileType 1`] = `"application"`;

exports[`system > 1211 > cron > noArgs 1`] = `"55 * 28 * 1"`;

exports[`system > 1211 > cron > with includeNonStandard false 1`] = `"55 * 28 * 1"`;

exports[`system > 1211 > cron > with includeNonStandard true 1`] = `"55 * 28 * 1"`;

exports[`system > 1211 > cron > with includeYear false 1`] = `"55 * 28 * 1"`;

exports[`system > 1211 > cron > with includeYear true 1`] = `"55 * 28 * 1 *"`;

exports[`system > 1211 > directoryPath 1`] = `"/var/log"`;

exports[`system > 1211 > fileExt > noArgs 1`] = `"dic"`;

exports[`system > 1211 > fileExt > with mimeType 1`] = `"map"`;

exports[`system > 1211 > fileName > noArgs 1`] = `"invoice_cyclocross_assault.z2"`;

exports[`system > 1211 > fileName > with extensionCount 1`] = `"invoice_cyclocross_assault.z2.tsd"`;

exports[`system > 1211 > filePath 1`] = `"/var/log/strictly_rustic.avi"`;

exports[`system > 1211 > fileType 1`] = `"x-shader"`;

exports[`system > 1211 > mimeType 1`] = `"text/vnd.dmclientscript"`;

exports[`system > 1211 > networkInterface > noArgs 1`] = `"wws8d1"`;

exports[`system > 1211 > networkInterface > with {"interfaceSchema":"index"} 1`] = `"wwo4"`;

exports[`system > 1211 > networkInterface > with {"interfaceSchema":"mac"} 1`] = `"wwx7ec32f0a2a3c"`;

exports[`system > 1211 > networkInterface > with {"interfaceSchema":"pci"} 1`] = `"P8wwp7s2f9d6"`;

exports[`system > 1211 > networkInterface > with {"interfaceSchema":"slot"} 1`] = `"wws4"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"en","interfaceSchema":"index"} 1`] = `"eno9"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"en","interfaceSchema":"mac"} 1`] = `"enxe7ec32f0a2a3"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"en","interfaceSchema":"pci"} 1`] = `"enp4s8d1"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"en","interfaceSchema":"slot"} 1`] = `"ens9f8"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"en"} 1`] = `"P8enp7s2f9d6"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"wl","interfaceSchema":"index"} 1`] = `"wlo9"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"wl","interfaceSchema":"mac"} 1`] = `"wlxe7ec32f0a2a3"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"wl","interfaceSchema":"pci"} 1`] = `"wlp4s8d1"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"wl","interfaceSchema":"slot"} 1`] = `"wls9f8"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"wl"} 1`] = `"P8wlp7s2f9d6"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"ww","interfaceSchema":"index"} 1`] = `"wwo9"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"ww","interfaceSchema":"mac"} 1`] = `"wwxe7ec32f0a2a3"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"ww","interfaceSchema":"pci"} 1`] = `"wwp4s8d1"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"ww","interfaceSchema":"slot"} 1`] = `"wws9f8"`;

exports[`system > 1211 > networkInterface > with {"interfaceType":"ww"} 1`] = `"P8wwp7s2f9d6"`;

exports[`system > 1211 > networkInterface > with {} 1`] = `"wws8d1"`;

exports[`system > 1211 > semver 1`] = `"9.4.8"`;

exports[`system > 1337 > commonFileExt 1`] = `"wav"`;

exports[`system > 1337 > commonFileName > noArgs 1`] = `"quia.mp2"`;

exports[`system > 1337 > commonFileName > with extension 1`] = `"quia.ext"`;

exports[`system > 1337 > commonFileType 1`] = `"audio"`;

exports[`system > 1337 > cron > noArgs 1`] = `"15 13 5 * *"`;

exports[`system > 1337 > cron > with includeNonStandard false 1`] = `"15 13 5 * *"`;

exports[`system > 1337 > cron > with includeNonStandard true 1`] = `"15 13 5 * *"`;

exports[`system > 1337 > cron > with includeYear false 1`] = `"15 13 5 * *"`;

exports[`system > 1337 > cron > with includeYear true 1`] = `"15 13 5 * * 2029"`;

exports[`system > 1337 > directoryPath 1`] = `"/Library"`;

exports[`system > 1337 > fileExt > noArgs 1`] = `"oa3"`;

exports[`system > 1337 > fileExt > with mimeType 1`] = `"json"`;

exports[`system > 1337 > fileName > noArgs 1`] = `"quia.cmc"`;

exports[`system > 1337 > fileName > with extensionCount 1`] = `"quia.cmc.kml"`;

exports[`system > 1337 > filePath 1`] = `"/Library/duane_fort.wmlc"`;

exports[`system > 1337 > fileType 1`] = `"font"`;

exports[`system > 1337 > mimeType 1`] = `"application/vnd.chipnuts.karaoke-mmd"`;

exports[`system > 1337 > networkInterface > noArgs 1`] = `"enx234870538945"`;

exports[`system > 1337 > networkInterface > with {"interfaceSchema":"index"} 1`] = `"eno5"`;

exports[`system > 1337 > networkInterface > with {"interfaceSchema":"mac"} 1`] = `"enx823487053894"`;

exports[`system > 1337 > networkInterface > with {"interfaceSchema":"pci"} 1`] = `"enp1s2f5d0"`;

exports[`system > 1337 > networkInterface > with {"interfaceSchema":"slot"} 1`] = `"ens5f2d5"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"en","interfaceSchema":"index"} 1`] = `"eno2"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"en","interfaceSchema":"mac"} 1`] = `"enx482348705389"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"en","interfaceSchema":"pci"} 1`] = `"P5enp1s2f5d0"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"en","interfaceSchema":"slot"} 1`] = `"ens2d2"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"en"} 1`] = `"ens5f2d5"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"wl","interfaceSchema":"index"} 1`] = `"wlo2"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"wl","interfaceSchema":"mac"} 1`] = `"wlx482348705389"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"wl","interfaceSchema":"pci"} 1`] = `"P5wlp1s2f5d0"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"wl","interfaceSchema":"slot"} 1`] = `"wls2d2"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"wl"} 1`] = `"wls5f2d5"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"ww","interfaceSchema":"index"} 1`] = `"wwo2"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"ww","interfaceSchema":"mac"} 1`] = `"wwx482348705389"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"ww","interfaceSchema":"pci"} 1`] = `"P5wwp1s2f5d0"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"ww","interfaceSchema":"slot"} 1`] = `"wws2d2"`;

exports[`system > 1337 > networkInterface > with {"interfaceType":"ww"} 1`] = `"wws5f2d5"`;

exports[`system > 1337 > networkInterface > with {} 1`] = `"enx234870538945"`;

exports[`system > 1337 > semver 1`] = `"2.5.1"`;

exports[`system > seed: 42 > commonFileExt() 1`] = `"png"`;

exports[`system > seed: 42 > commonFileName() 1`] = `"lavender_shoes.mpe"`;

exports[`system > seed: 42 > commonFileType() 1`] = `"audio"`;

exports[`system > seed: 42 > cron() 1`] = `"* 19 * 3 5"`;

exports[`system > seed: 42 > directoryPath() 1`] = `"/opt/bin"`;

exports[`system > seed: 42 > fileExt() 1`] = `"lrm"`;

exports[`system > seed: 42 > fileName() 1`] = `"lavender_shoes.jxsc"`;

exports[`system > seed: 42 > filePath() 1`] = `"/opt/bin/comeback_neptunium_hip.ptid"`;

exports[`system > seed: 42 > fileType() 1`] = `"image"`;

exports[`system > seed: 42 > mimeType() 1`] = `"application/vnd.ibm.rights-management"`;

exports[`system > seed: 42 > networkInterface() 1`] = `"wlp1s7"`;

exports[`system > seed: 42 > semver() 1`] = `"3.7.9"`;

exports[`system > seed: 1211 > commonFileExt() 1`] = `"htm"`;

exports[`system > seed: 1211 > commonFileName() 1`] = `"invoice_cyclocross_assault.mp4"`;

exports[`system > seed: 1211 > commonFileType() 1`] = `"application"`;

exports[`system > seed: 1211 > cron() 1`] = `"55 * 28 * 1"`;

exports[`system > seed: 1211 > directoryPath() 1`] = `"/var/log"`;

exports[`system > seed: 1211 > fileExt() 1`] = `"dic"`;

exports[`system > seed: 1211 > fileName() 1`] = `"invoice_cyclocross_assault.z2"`;

exports[`system > seed: 1211 > filePath() 1`] = `"/var/log/strictly_rustic.avi"`;

exports[`system > seed: 1211 > fileType() 1`] = `"x-shader"`;

exports[`system > seed: 1211 > mimeType() 1`] = `"text/vnd.dmclientscript"`;

exports[`system > seed: 1211 > networkInterface() 1`] = `"wws8d1"`;

exports[`system > seed: 1211 > semver() 1`] = `"9.4.8"`;

exports[`system > seed: 1337 > commonFileExt() 1`] = `"wav"`;

exports[`system > seed: 1337 > commonFileName() 1`] = `"quia.mp2"`;

exports[`system > seed: 1337 > commonFileType() 1`] = `"audio"`;

exports[`system > seed: 1337 > cron() 1`] = `"15 13 5 * *"`;

exports[`system > seed: 1337 > directoryPath() 1`] = `"/Library"`;

exports[`system > seed: 1337 > fileExt() 1`] = `"oa3"`;

exports[`system > seed: 1337 > fileName() 1`] = `"quia.cmc"`;

exports[`system > seed: 1337 > filePath() 1`] = `"/Library/duane_fort.wmlc"`;

exports[`system > seed: 1337 > fileType() 1`] = `"font"`;

exports[`system > seed: 1337 > mimeType() 1`] = `"application/vnd.chipnuts.karaoke-mmd"`;

exports[`system > seed: 1337 > networkInterface() 1`] = `"enx234870538945"`;

exports[`system > seed: 1337 > semver() 1`] = `"2.5.1"`;
